orchestrator/utils/params/params.go

```diff
@@ -0,0 +1,46 @@
+package params
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+
+	"github.com/SayedAlesawy/Videra-Ingestion/orchestrator/utils/errors"
+)
+
+// logPrefix Used for hierarchical logging
+var logPrefix = "[Params-Manager]"
+
+// orchestratorParamsOnce Used to garauntee thread safety for singleton instances
+var orchestratorParamsOnce sync.Once
+
+// orchestratorParamsInstance A singleton instance of the orchestrator params object
+var orchestratorParamsInstance *OrchestratorParams
+
+// OrchestratorParamsInstance A function to return the orchestrator params singleton instance
+func OrchestratorParamsInstance() *OrchestratorParams {
+	orchestratorParamsOnce.Do(func() {
+		params := os.Args
+
+		orchestratorParams := OrchestratorParams{
+			VideoPath:       params[1],
+			ModelPath:       params[2],
+			ModelConfigPath: params[3],
+			StartIdx:        intParam(params[4]),
+			FrameCount:      intParam(params[5]),
+		}
+
```

Comment:
we could add validation that paths exits (videoPath and ModelPath)

Reply:
Okay, that's a good idea.

Reply:
Done.